Allow edit only the main email template

costasovo opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR introduces "Edit template" button that allows users to open the main block email template used by all transactional emails in the email editor directly from the settings. Previously they needed to open an individual email and then access the editor. 
![Screenshot 2025-04-14 at 16 49 51](https://github.com/user-attachments/assets/769da147-8f60-432b-b5ab-6018c64b0496)



<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

#### Changes
- Email editor page loader newly checks the template query parameter, and in case it is present, it attempts to open the editor with the template as the main and only document.
- A couple of changes in the editor applications were needed to support the template mode. Visible changes are related to saving buttons. When template mode is active, we don't display "save draft" and we display the main save button covering all documents.
- "Edit template" button was added to the email settings page

Note: We currently still need to pass the post_id of an email post to fulfill conditions for checks in post.php. We may consider fixing this by moving the editor to a different URL.

Closes WOOPLUG-3312 .

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Activate the block email editor feature (Settings > Advanced > Features > Experimental)
2. Go to Settings > Emails and click the "Edit template" button and test template editing, saving changes, etc.
3. Go to Settings > Emails and open a couple of individual emails. Verify that changes done in the template mode are present and that the editor for individual emails works as expected.

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

costasovo committed:
Update email editor page renderer also to support template

When a template is present in the query parameters, we ignore post
params that are present only to pass conditions on post.php, and
we setup the editor to use the template as the primary document.
WOOPLUG-3312

costasovo committed:
Update the email editor to work only with a template

- template uses string IDs so we need to allow them
- we need a few adjustments in header so that save buttons make sense
WOOPLUG-3312

costasovo committed:
Move email sectio description to react

This change will make it easier to add the edit template button.
WOOPLUG-3312

costasovo committed:
Add Edit template button to the email settings page

We need to pass a post ID to get over checks in post.php. I used
post from one of the emails from the listing.
WOOPLUG-3312

costasovo committed:
Add changelogs

WOOPLUG-3312

lysyjan reviewed (commented):
The code looks good to me, but I have just noticed some things to discuss during my testing.
1. I was not able to send a preview to my email due to an error. Maybe we can hide the send option here.
2. It looks a bit confusing to me that I cannot return to the email settings.

@costasovo What do you think about it?

costasovo committed:
Reintroduce the Save email button in template mode to return back to settings

WOOPLUG-3312

costasovo committed:
Hide irrelevant preview options for template mode

WOOPLUG-3312

costasovo commented:
> The code looks good to me, but I have just noticed some things to discuss during my testing.
> 
>  - I was not able to send a preview to my email due to an error. Maybe we can hide the send option here.
>  - It looks a bit confusing to me that I cannot return to the email settings.

@lysyjan I addressed the issues in the last two commits.

lysyjan reviewed (approved):
@costasovo Thanks. It works as expected for me now.

Here is a video where I tested the new feature:
[Screen Recording at 2025-04-22 at 11-24-42.webm](https://github.com/user-attachments/assets/891b0fca-eb95-468c-89c7-f486e248d1a9)
